src/status_im/contexts/wallet/collectible/events.cljs

```diff
@@ -83,81 +83,183 @@
+               {:address             address
+                :reset-collectibles? true}]])]})))
+
+(rf/reg-event-fx
+ :wallet/collectibles-data-updated
+ (fn [{:keys [db]} [{:keys [message]}]]
+   (let [collectibles-by-address           (->> message
+                                                transforms/json->clj
+                                                data-store/rpc->collectibles
+                                                (group-by #(-> % :ownership first :address)))
+         update-collectibles-in-account-fn (fn [existing-collectibles updated-collectibles]
+                                             (let [idxd-existing (utils.collection/index-by
+                                                                  :unique-id
+                                                                  existing-collectibles)
+                                                   existing-ids  (-> idxd-existing keys vec)
+                                                   ;; pick collectibles only in the app-db
+                                                   idxd-updated  (-> (utils.collection/index-by
+                                                                      :unique-id
+                                                                      updated-collectibles)
+                                                                     (select-keys existing-ids))]
+                                               (-> (merge-with
+                                                    merge
+                                                    idxd-existing
+                                                    idxd-updated)
+                                                   vals
+                                                   vec)))]
+     {:db (update-in db
+                     [:wallet :accounts]
+                     #(reduce-kv
+                       (fn [accounts address updated-collectibles]
+                         (if (contains? accounts address)
+                           (update-in accounts
+                                      [address :collectibles]
+                                      update-collectibles-in-account-fn
+                                      updated-collectibles)
+                           accounts))
+                       %
+                       collectibles-by-address))})))
+
+(rf/reg-event-fx
+ :wallet/set-collectibles-updating-status
+ (fn [{:keys [db]} [address chain-ids]]
+   {:db (assoc-in db [:wallet :ui :collectibles :updating address] chain-ids)}))
 
 (defn- update-fetched-collectibles-progress
-  [db owner-address collectibles offset has-more?]
+  [db owner-address collectibles offset has-more? reset-collectibles?]
   (-> db
-      (assoc-in [:wallet :ui :collectibles :fetched owner-address] collectibles)
+      (update-in [:wallet :accounts owner-address :collectibles]
+                 #(if reset-collectibles? collectibles (reduce conj (or % []) collectibles)))
```

Comment:
btw, sometimes I feel we abuse of the one-liner `if` statement, it's usually clearer if we add a new line when complex  s-forms appear, I personally try to only use it as a one-liner when the forms are very simple.

I mean this one reads better:
```clojure
(if reset-collectibles?
  collectibles
  (reduce ... collectibles))
```